src/plugins/janus_videoroom.c

```diff
@@ -6866,6 +6866,19 @@ void janus_videoroom_setup_media(janus_plugin_session *handle) {
 			/* Notify all other participants that there's a new boy in town */
 			janus_videoroom_notify_about_publisher(participant, FALSE);
 			janus_refcount_decrease(&participant->ref);
```

Comment:
You should move all your new code before this line, that is before we decrease the participant's ref. For all we know, the participant may be destroyed after this point.